update/v1alpha1/types_update_status.go

```diff
@@ -0,0 +1,750 @@
+	// (to the original version) was not fully completed
+	PartialMetadata VersionMetadataKey = "partial"
+	// architecture denotes a string that indicates the architecture of the payload image of the version,
+	// when relevant
+	ArchitectureMetadata VersionMetadataKey = "architecture"
+)
+
+type VersionMetadata struct {
+	// +required
+	Key VersionMetadataKey `json:"key"`
+
+	// +required
+	VersionMetadataValue `json:",inline"`
+}
+
+type VersionMetadataValue struct {
+	// +unionDiscriminator
+	// +required
+	// +kubebuilder:validation:Enum:string;bool;int
+	Type VersionMetadataType `json:"type"`
+
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +unionMember
+	String string `json:"string,omitempty"`
+
+	// +optional
+	// +kubebuilder:validation:Type=bool
+	// +unionMember
+	Bool bool `json:"bool,omitempty"`
+
+	// +optional
+	// +kubebuilder:validation:Type=int
+	// +unionMember
+	Integer int `json:"integer,omitempty"`
+}
+
+// Version describes a version involved in an update, typically on one side of an update edge
+type Version struct {
+	// version is a semantic version string
+	// +required
+	Version string `json:"version,omitempty"`
+
+	// metadata is a list of metadata associated with the version
+	// +listType=map
+	// +listMapKey=key
+	// +optional
+	Metadata []VersionMetadata `json:"metadata,omitempty"`
+}
+
+// ControlPlaneUpdateVersions contains the original and target versions of the upgrade
+type ControlPlaneUpdateVersions struct {
+	// previous is the version of the control plane before the update
+	// +optional
+	Previous Version `json:"previous,omitempty"`
+
+	// target is the version of the control plane after the update
+	// +required
+	Target Version `json:"target"`
+}
+
+// ClusterVersionStatusInsight reports the state of a ClusterVersion resource (which represents a control plane
+// update in standalone clusters), during the update.
+type ClusterVersionStatusInsight struct {
+	// resource is the ClusterVersion resource that represents the control plane
+	// +required
+	Resource ResourceRef `json:"resource"`
+
+	// assessment is the assessment of the control plane update process
+	// +required
+	Assessment ControlPlaneAssessment `json:"assessment"`
+
+	// versions contains the original and target versions of the upgrade
+	// +required
+	Versions ControlPlaneUpdateVersions `json:"versions"`
+
+	// completion is a percentage of the update completion (0-100)
+	// +required
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	Completion uint8 `json:"completion"`
+
+	// startedAt is the time when the update started
+	// +required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	StartedAt metav1.Time `json:"startedAt"`
+
+	// completedAt is the time when the update completed
+	// +optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=date-time
+	CompletedAt metav1.Time `json:"completedAt,omitempty"`
```

Comment:
Needs to be a `*metav1.Time` to have `omitempty` work, see [my comment here][1], [this comment upstream][2], and [ClusterVersion `history` precedent][3].

[1]: https://github.com/openshift/api/pull/2012#discussion_r1790955342
[2]: https://github.com/golang/go/issues/51261#issuecomment-1048221441
[3]: https://github.com/openshift/api/blob/82e082220d910f489d35880174e2eb90e21f5589/config/v1/types_cluster_version.go#L245